src/common/policy.c

```diff
@@ -107,10 +107,8 @@ daos_policy_try_parse(const char *policy_str, struct policy_desc_t *out_pd)
 		/* tokenize left and right items of "name=value" */
 		str2 = tok;
 
-		for (j = 0; ; j++, str2 = NULL) {
-			tok = strtok_r(str2, delim2, &saveptr2);
-			if (tok == NULL)
-				break;
+		while ((tok = strtok_r(str2, delim2, &saveptr2)) != NULL) {
```

Comment:
minimal fix would be `for ( ; ; str2 = NULL) {` but what you have looks fine

Reply:
Yes, I tried that first but really this code should be using a while loop but was using a for loop.

Reply:
If c didn't have while loops and we wanted to simulate one then yes I'd agree that would be sensible.  Let's use the language as it's designed however.